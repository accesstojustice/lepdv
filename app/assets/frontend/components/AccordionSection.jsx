import { Component } from 'react'
import { Button, Panel } from 'react-bootstrap'
import PrintableRequest from './PrintableRequest'

export default class AccordionSection extends Component {
  constructor(...args) {
    super(...args);
    this.state = {
      open: false
    };
  }

  render() {
    return (
      <div>

        <h3 className="center block">Resource Options</h3>
        <Button
          bsStyle="primary"
          block
          onClick={ ()=> this.setState({ open: !this.state.open })}>Multilingual Safety Plans
        </Button>
        <Panel collapsible expanded={this.state.open}>
          <ul>
            <li><strong>Introduction: </strong>You cannot control your (ex) partnerâ€™s violence, but it may be possible to increase your own safety as well as the safety of your children</li>
            <li><strong>Taking Care of Yourself: </strong>Take part in social activities (e.g. programs at your local community or Early Years centre, dinner with friends, etc.)</li>
            <li><strong>Escape Plan: </strong>Make Copies of:  Marriage certificate, divorce papers, custody documentation, court orders, restraining orders, or other legal documents; Work Permits; Bank Books</li>
          </ul>
          <a href='http://www.pcawa.net/uploads/1/4/7/1/14711308/pcawa_safety_handbook_2010_10_28_-_eng.pdf'>
            <div className="pull-right">
              <Button className="source" bsStyle="default" bsSize="xsmall">- source</Button>
            </div>
          </a>

        </Panel>

        <Button
          bsStyle="primary"
          block
          onClick={ ()=> this.setState({ open: !this.state.open })}>Right to an Interpreter
        </Button>
        <Panel collapsible expanded={this.state.open}>
          <ul>
            <li>Although the United States Constitution does not explicitly provide for the right to an interpreter, the individual rights and liberties afforded to all individuals under the Fourth, Fifth, Sixth, Eighth and Fourteenth Amendments are meaningless for non- or limited-English speakers unless they are provided with complete, competent, and accurate interpreting services.</li>
            <li>In recent years, the U.S. Department of Justice has been reviewing state court's compliance with this executive order.</li>
          </ul>

          <a href='http://www.najit.org/certification/faq.php'>
            <div className="pull-right">
              <Button className="source" bsStyle="default" bsSize="xsmall">- source</Button>
            </div>
          </a>
          <PrintableRequest />


        </Panel>

        <Button bsStyle="primary"
          block
          onClick={ () => this.setState({ open: !this.state.open })}>Interpretation Contact
        </Button>
        <Panel collapsible expanded={this.state.open}>
          <ul>
            <li>Describe your roles and clarify expectations</li>
            <li>Ensure that the interpreter is taking notes</li>
            <li>Pause frequently to allow the interpreter to translate information</li>
          </ul>
          {/* // TODO: no need to link to addresses+ info on all interpreters
            <a href='https://publicapps.illinoiscourts.gov/pub/f?p=303:25:'>
            <div className="pull-right">
            <Button className="source" bsStyle="success" bsSize="xsmall">interpreter registry</Button>
            </div>
            </a>
            */}
            <a href='http://www.asha.org/practice/multicultural/issues/interpret.htm'>
              <div className="pull-right">
                <Button className="source" bsStyle="default" bsSize="xsmall">- source</Button>
              </div>
            </a>

          </Panel>

          <Button
            bsStyle="primary"
            block
            onClick={ () => this.setState({ open: !this.state.open })}>Legal Terms
          </Button>
          <Panel collapsible expanded={this.state.open}>
            <ul>
              <li>Words and phrases used in the courtroom and conversational-English are <strong>not</strong> the same.</li>
              <li>Note: translations are various and my differ, according to regional dialects of target (foreign) language used, etc.</li>
              <li><em>Attention: </em>Children of the victim, or the offending party -- <strong>do not make adequate interpreters.</strong></li>
            </ul>
            {/* // TODO: link to an interpreter resource that helps decifer phrases, or glossaries
              <a href='https://publicapps.illinoiscourts.gov/pub/f?p=303:25:'>
              <div className="pull-right">
              <Button className="source" bsStyle="success" bsSize="xsmall">interpreter registry</Button>
              </div>
              </a>
              */}
              <a href='http://www.abajournal.com/magazine/article/justice_moves_slowly_for_those_who_need_interpreters'>
                <div className="pull-right">
                  <Button className="source" bsStyle="warning" bsSize="xsmall">more on this topic</Button>
                </div>
              </a>
            </Panel>

            <Button
              bsStyle="primary"
              block
              onClick={ () => this.setState({ open: !this.state.open })}>I'm Going to Court
            </Button>
            <Panel collapsible expanded={this.state.open}>
              <ul>
                <li>You cannot contact the office of interpreter services directly.</li>
              </ul>
              <a href='http://www.cookcountycourt.org/ABOUTTHECOURT/OfficeoftheChiefJudge/CourtRelatedServices/OfficeofInterpreterServices.aspx'>
                <div className="pull-right">
                  <Button className="source" bsStyle="default" bsSize="xsmall">- source</Button>
                </div>
              </a>
              <PrintableRequest />

            </Panel>
          </div>
        )
      }
    }
