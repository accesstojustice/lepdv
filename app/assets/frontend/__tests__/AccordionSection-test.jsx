// workaround for 'being called-twice' error
jest.mock('react/lib/ReactDefaultInjection');

var React = require('react');

import AccordionSection from '../components/AccordionSection';
import renderer from 'react-test-renderer';

describe('Accordion Section', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <AccordionSection open={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();

  });


  it('renders an expanded accordion', () => {
    const component = renderer.create(
      <AccordionSection open={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
        console.log(3, tree);

    // // manually trigger the callback
    // tree.props.toggleOpen()
    //
    //
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    //
    // // manually trigger the callback
    // tree.props.toggleOpen();
    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
  });
});
