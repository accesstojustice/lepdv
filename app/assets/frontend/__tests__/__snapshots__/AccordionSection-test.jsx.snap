exports[`Accordion Section expands the accordion when clicked 1`] = `
<div>
  <h3
    className="center block">
    Resource Options
  </h3>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Multilingual Safety Plans
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Right to an Interpreter
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Interpretation Contact
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Legal Terms
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                more on this topic
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      I'm Going to Court
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Section renders an expanded accordion 1`] = `
<div>
  <h3
    className="center block">
    Resource Options
  </h3>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Multilingual Safety Plans
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={false}
        className="panel-collapse collapse in"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Right to an Interpreter
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={false}
        className="panel-collapse collapse in"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Interpretation Contact
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={false}
        className="panel-collapse collapse in"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Legal Terms
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={false}
        className="panel-collapse collapse in"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                more on this topic
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      I\'m Going to Court
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={false}
        className="panel-collapse collapse in"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion Section renders correctly 1`] = `
<div>
  <h3
    className="center block">
    Resource Options
  </h3>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Multilingual Safety Plans
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Right to an Interpreter
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Interpretation Contact
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      Legal Terms
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                more on this topic
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={undefined}
      type="button">
      I\'m Going to Court
    </button>
    <div
      className="panel panel-default"
      id={null}>
      <div
        aria-expanded={null}
        aria-hidden={true}
        className="panel-collapse collapse"
        id={undefined}
        role={undefined}>
        <div
          className="panel-body">
          <ul>
            <li>
              <strong />
            </li>
          </ul>
          <a
            href={undefined}>
            <div
              className="pull-right">
              <button
                className="source btn btn-xs btn-default"
                disabled={false}
                type="button">
                - source
              </button>
            </div>
          </a>
          <div
            className="print-tip center">
            <a
              href="http://12.218.239.52/Forms/pdf_files/CCG0108.pdf">
              <small>
                <strong>
                  Affirmatively
                </strong>
                 ask for an interpreter.
              </small>
              <div
                className="pull-left">
                <button
                  className="source btn btn-xs btn-warning"
                  disabled={false}
                  type="button">
                  printable form
                </button>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
